client/src/components/AddAgent/index.js

```diff
@@ -1,17 +1,83 @@
-import React, { useState } from 'react';
-
+import React, { useState, useContext } from 'react';
+import { Link, useHistory } from "react-router-dom";
+import AuthContext from '../../utils/AuthContext';
+import Errors from "../Errors";
 
 import './style.css'
 
 
 const AddAgent = (props) => {
 
-    
+    const [agent, setAgent] = useState({
+        firstName: "",
+        middleName: "",
+        lastName: "",
+        dob: "",
+        heightInInches: 36,
+    });
+
+    const [errors, setErrors] = useState([]);
+
+    const auth = useContext(AuthContext);
+    const history = useHistory();
+
+    const inputChangeHandler = (e) => {
+        setAgent({
+            ...agent,
+            [e.target.name]: e.target.value
+        })
+    };
+
+    const handleSubmitAddForm = (e) => {
+        e.preventDefault();
+
+        const newAgent = {
+            firstName: e.target.firstName.value,
+            lastName: e.target.lastName.value,
+            middleName: e.target.middleName.value,
+            dob: e.target.dob.value,
+            heightInInches: e.target.heightInInches.value
+        };
```

Comment:
Since agent is an object, you can put `newagent = { ...agent }` here. A small change but could save you some typing!

Reply:
You're setting all those values already in your change handler